src/main/components/RequestBar.jsx

```diff
@@ -9,36 +10,50 @@ const RequestBar = (props) => {
   const [selected, setSelected] = useState(method);
   const [uri, setUri] = useState('');
 
+  // header info
+  const [headerType, setHeaderType] = useState('Authorization');
+  const [authType, setType] = useState('Bearer Token');
+  const [headerKey, setHeaderKey] = useState('');
 
   const handleChange = (e) => {
     const { name, value } = e.target;
     if (name === 'method') setSelected(value);
     else if (name === 'uri') setUri(value);
+
+    // header info
+    if (name === 'header') setHeaderType(value);
+    if (name === 'headerKey') setHeaderKey(`Bearer ${value}`);
+
+  };
+
+  const runTest = (link, sendingObj, testsClone, i) => {
+    const test = testsClone;
```

Comment:
I don't think this duplicates `testsClone`-- now `tests` is just a reference to testsClone.  Not a prob bc presumably it isn't modifying state considering it's called "clone", just worth noting

Reply:
I agree its point to a reference in memory, but for cleanliness we I did not want to mutate directly from a parameter passed in.